### PR TITLE
Refactor Enterprise Beans no-interface view bytecode generation

### PR DESCRIPTION
**Pull request by avpinchuk**
What's done:

* Make code more stylistically  consistent;
* Remove code duplicates;
* General code cleanup;
* Cleanup generated bytecode;
* Fix deployment bug. It could happen when session bean does not have the no-args constructor but has a CDI `@Inject`-annotated constructor and one one more "regular" constructors with primitive arguments (it's not prohibited by CDI spec explicitly).

Signed-off-by: Alexander Pinčuk <alexander.v.pinchuk@gmail.com>

### CODE REVIEW

**Comment by pzygielo** [comment]
I'd rather see the fix as separate PR, not mixed into refactoring one.

File: appserver/ejb/ejb-full-container/src/main/java/org/glassfish/ejb/mdb/MessageBeanInterfaceGenerator.java [hunk 1]
```diff
@@ -16,27 +17,29 @@
 
 package org.glassfish.ejb.mdb;
 
-import com.sun.ejb.codegen.EjbOptionalIntfGenerator;
```
Comment (dmatej) [inline appserver/ejb/ejb-full-container/src/main/java/org/glassfish/ejb/mdb/MessageBeanInterfaceGenerator.java:19]: Ordering imports - alphabetical.

**Comment by avpinchuk** [comment]
@pzygielo, I'll split this PR to two ones. Now I close this.

File: appserver/ejb/ejb-container/src/main/java/com/sun/ejb/codegen/EjbOptionalIntfGenerator.java [hunk 2]
```diff
@@ -265,157 +213,154 @@ public void generateSubclass(Class superClass, String subClassName, Class delega
...
-        java.lang.reflect.Method mth = Object.class.getDeclaredMethod("toString");
-        if (!hasSameSignatureAsExisting(mth, allMethods)) {
-            generateToStringBeanMethod(tv, superClass);
+        try {
+            Method method = Object.class.getDeclaredMethod("toString");
+            if (!hasSameSignatureAsExisting(method, allMethods)) {
+                generateToStringBeanMethod(cw, superClass);
+            }
+        } catch (NoSuchMethodException ignored) {
+            // Should never be reached
```
Comment (dmatej) [inline appserver/ejb/ejb-container/src/main/java/com/sun/ejb/codegen/EjbOptionalIntfGenerator.java:230]: If it should never be reached, then better throw IllegalStateException, but don't hide it.

File: appserver/ejb/ejb-full-container/src/main/java/org/glassfish/ejb/mdb/MessageBeanInterfaceGenerator.java [hunk 3]
```diff
@@ -16,27 +17,29 @@
...
 
+import com.sun.ejb.codegen.EjbOptionalIntfGenerator;
+
 public class MessageBeanInterfaceGenerator extends EjbOptionalIntfGenerator {
 
     public MessageBeanInterfaceGenerator(ClassLoader loader) {
         super(loader);
     }
 
-    public Class generateMessageBeanSubClass(Class<?> beanClass, Class messageBeanInterface) throws Exception {
```
Comment (dmatej) [inline appserver/ejb/ejb-full-container/src/main/java/org/glassfish/ejb/mdb/MessageBeanInterfaceGenerator.java:29]: Could we limit it somehow, like specify <T> Class<? extends T> gmbs(Class<T> . Not sure, if it is not possible, let it be, it is just an idea.

**Review by dmatej — COMMENTED** [review]
Nevermind, I will use this PR to run the TCK. This part is quite fragile, "caller-sensitive", it has to respect JPMS, MethodHandler rules, etc.

**Comment by dmatej** [comment]
The TCK: https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck/job/10.0.x/

**Pull request closed without merging**